include/manager.hpp

```diff
@@ -154,6 +185,38 @@ class Manager
      */
     void performVPDRecollection();
 
+    /**
+     * @brief Get un-expanded location code.
+     * An API to split expanded location code to its un-expanded
+     * format and node number.
+     *
+     * @param[in] i_locationCode - Location code in expanded format.
+     *
+     * @return Location code in un-expanded format and its node number.
+     */
+    std::tuple<std::string, uint16_t>
+        getCollapsedLocationCode(const std::string& i_locationCode);
+
+    /**
+     * @brief An api to check validity of location code
+     *
+     * @param[in] i_locationCode - location code
+     * @return true/false based on validity check
+     */
+    bool isValidLocationCode(const std::string i_locationCode)
```

Comment:
Why is this public?
Also if its not just one line or just a return, better to keep definition in Cpp file only. Keeps the header clean.

Reply:
Updated